lib/matplotlib/backends/backend_svg.py

```diff
@@ -125,17 +208,29 @@ def __flush(self, indent=True):
             self.__write(escape_cdata(data))
             self.__data = []
 
-    ## Opens a new element.  Attributes can be given as keyword
-    # arguments, or as a string/string dictionary. The method returns
-    # an opaque identifier that can be passed to the <b>close</b>
-    # method, to close all open elements up to and including this one.
-    #
-    # @param tag Element tag.
-    # @param attrib Attribute dictionary.  Alternatively, attributes
-    #    can be given as keyword arguments.
-    # @return An element identifier.
-
     def start(self, tag, attrib={}, **extra):
+        """Opens a new element.  Attributes can be given as keyword
+        arguments, or as a string/string dictionary. The method returns
+        an opaque identifier that can be passed to the `close` method,
+        to close all open elements up to and including this one.
+
+        This method flushes the internal buffer before it begins making
+        additions.
+
+        Parameters
+        ----------
+        tag : str
+            The element name to open
+        attrib : dict, optional
+            A mapping of name=value pairs to be included in the opened tag
+        **extra
+            Any additional name=value pairs not part of `attrib`
+        
+        Returns
+        -------
+        int
+            The size of the :attr:`__tags`, the current tag stack, - 1
```

Comment:
The only place this is actually used either as an element `id` attribute or even assigned is when creating the topmost `<svg>`. In all other cases, the return value of `XMLWriter.start` is ignored.

The id attribute of artist-mapped elements are set in `RendererSVG.open_group`, which either uses the user-provided `gid` or is pulled from `RendererSVG._groupd`, which is treated like a `Counter` mapping `artist_type` to `count`.  Even using it as an id is questionable because the stack grows and shrinks as the XML tree grows.